File: backend/models/user/UserModel.cs
namespace User.Models;

public class UserModel
{
    public UserModel()
    {
    }

    public UserModel(string name, string email, string password)
    {
        Name = name;
        Email = email;
        Password = password;
        Id = Guid.NewGuid();
    }

    [Key]
    public Guid Id { get; init; }

    [Required]
    public string Name { get; set; }
    public string Email { get; set; }
    public string Password { get; set; }

}

File: moviequest/lib/models/user_model.dart
import 'dart:convert';
import 'package:json_annotation/json_annotation.dart';

part 'user_model.g.dart';

@JsonSerializable()
class UserModel {
  String? name;
  String? email;
  String? token;
  String? password;

  UserModel({
    this.name,
    this.email,
    this.token,
    this.password,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'name': name,
      'email': email,
      'toke': token,
      'password': password,
    };
  }

  factory UserModel.fromMap(Map<String, dynamic> map) {
    return UserModel(
      name: map['Name'] as String,
      email: map['Email'] as String,
      token: map['Token'] as String,
    );
  }

  String toJson() => json.encode(toMap());

  factory UserModel.fromJson(String source) =>
      UserModel.fromMap(json.decode(source) as Map<String, dynamic>);
}
